src/VariantViewer.js

```diff
@@ -32,21 +54,47 @@ var variantsFill = function(d, fv) {
             return LegendDialog.UPNonDiseaseColor;
         }
     } else {
-        var sift = false, polyphen = false;
-        if ((d.polyphenPrediction !== undefined) && (d.polyphenPrediction !== 'unknown')) {
-            polyphen = d.polyphenScore !== undefined ? true : false;
+        var externalPrediction, extDatum = {};
+        if (d.externalData) {
+            var keys = _.keys(d.externalData);
+            extDatum = d.externalData[keys[0]];
+            externalPrediction = getPredictionColorScore(extDatum.siftScore, extDatum.siftPrediction,
+                extDatum.polyphenScore, extDatum.polyphenPrediction);
+            extDatum.siftInUse = false;
+            extDatum.polyphenInUse = false;
         }
-        if (d.siftPrediction !== undefined) {
-            sift = d.siftScore !== undefined ? true : false;
+        var predicitonScore = getPredictionColorScore(d.siftScore, d.siftPrediction, d.polyphenScore,
+            d.polyphenPrediction);
+
+        if (fv.overwritePredictions === true) {
```

Comment:
Mmm, do you mean adding a function that takes care of the code inside the conditionals? Or do you refer to some method chaining?